### PR TITLE
Feat/rule detection

### PR DESCRIPTION
**Pull request by dwertent**
## Overview

___

## **Type**
enhancement, bug_fix

___

## **Description**
- Introduced a new CRD for runtime rule alert bindings.
- Added support for ClamAV including configuration files and related Kubernetes resources.
- Updated ClusterRoles, ConfigMaps, and DaemonSets to support new features like runtime detection and malware detection.
- Removed outdated sections from README and updated viewing instructions.
- Updated synchronizer configuration to support runtime alerts.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Configuration changes</strong></td><td><details><summary>4 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>_helpers.tpl</strong><dd><code>Add ClamAV Configuration Support</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

charts/kubescape-operator/templates/_helpers.tpl
<li>Added <code>clamAV</code> configuration to enable/disable ClamAV based on <br><code>capabilities.malwareDetection</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/kubescape/helm-charts/pull/416/files#diff-aa81f2ba9ca60237f5e62c64faf4e05f73d0f94f6ed0b9773aed5a84aedf993c">+3/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>clamd.conf</strong><dd><code>Add ClamAV Daemon Configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

charts/kubescape-operator/clamav/clamd.conf
- Added ClamAV daemon configuration file.

</details>
    

  </td>
  <td><a href="https://github.com/kubescape/helm-charts/pull/416/files#diff-2be930a130ebf176fc89c7911c7f0b6ea8077b40f01f97ec59157b635c7db9e6">+13/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>freshclam.conf</strong><dd><code>Add ClamAV Freshclam Configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

charts/kubescape-operator/clamav/freshclam.conf
- Added ClamAV freshclam configuration file.

</details>
    

  </td>
  <td><a href="https://github.com/kubescape/helm-charts/pull/416/files#diff-da734242d078156710b1f719cd46181dedaedd525d7fa4a7ac55fe2e1839ef70">+16/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>values.yaml</strong><dd><code>Enable New Capabilities and Update Configurations</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

charts/kubescape-operator/values.yaml
<li>Enabled new capabilities for runtime detection and malware detection.<br> <li> Updated configurations for node-agent and added ClamAV configurations.<br> <br>

</details>
    

  </td>
  <td><a href="https://github.com/kubescape/helm-charts/pull/416/files#diff-2d5611f69251d498d71f52fa778f6ce1bc93e1e1f46951ede1c50d975bdcad02">+58/-3</a>&nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>README.md</strong><dd><code>Update README with Simplified Viewing Instructions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

charts/kubescape-operator/README.md
<li>Removed sections for viewing scan summaries.<br> <li> Updated instructions for viewing configuration scan reports and image <br>vulnerabilities.<br>

</details>
    

  </td>
  <td><a href="https://github.com/kubescape/helm-charts/pull/416/files#diff-29301ee79c4acf85dcfdaa8326521bc7676bd383dff72b2e41a08c86b6727327">+2/-12</a>&nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr><tr><td><strong>Enhancement</strong></td><td><details><summary>8 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>runtime-rule-binding.crd.yaml</strong><dd><code>Introduce Runtime Rule Alert Binding CRD</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

charts/kubescape-operator/crds/runtime-rule-binding.crd.yaml
- Introduced a new CRD for runtime rule alert bindings.

</details>
    

  </td>
  <td><a href="https://github.com/kubescape/helm-charts/pull/416/files#diff-33c9ca5fc41241a81bf0fbb6a13314e29b628d0e84b073e50077cc0661b6bf18">+161/-0</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>clusterrole.yaml</strong><dd><code>Expand Node-Agent ClusterRole Permissions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

charts/kubescape-operator/templates/node-agent/clusterrole.yaml
- Added `namespaces` and `runtimerulealertbindings` to resources.

</details>
    

  </td>
  <td><a href="https://github.com/kubescape/helm-charts/pull/416/files#diff-d1c6835d60d9da525a40c8bb19251208521160b7bf07a5f57baeaec508a97f8e">+4/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>configmap.yaml</strong><dd><code>Update Node-Agent ConfigMap with New Features</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

charts/kubescape-operator/templates/node-agent/configmap.yaml
<li>Added configurations for runtime detection, network service, and <br>ClamAV.<br> <li> Introduced a new ConfigMap for ClamAV if enabled.<br>

</details>
    

  </td>
  <td><a href="https://github.com/kubescape/helm-charts/pull/416/files#diff-b8ccd25dcf7198d49ac143cb1ab55a34bc09894349c91c3d6767ba9ca210094d">+19/-1</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>daemonset.yaml</strong><dd><code>Update Node-Agent DaemonSet for ClamAV Support</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

charts/kubescape-operator/templates/node-agent/daemonset.yaml
<li>Added a new container for ClamAV if enabled.<br> <li> Updated environment variables and volume mounts for node-agent.<br>

</details>
    

  </td>
  <td><a href="https://github.com/kubescape/helm-charts/pull/416/files#diff-87cf3c6035cbedc17bdc75a7055fa64e7e38d2e83f4b0a79654a4c37c94f4efd">+35/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>default-rule-binding.yaml</strong><dd><code>Add Default Rule Binding for Runtime Detection</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

charts/kubescape-operator/templates/node-agent/default-rule-binding.yaml
- Added a default rule binding for runtime detection.

</details>
    

  </td>
  <td><a href="https://github.com/kubescape/helm-charts/pull/416/files#diff-db8a14b545acfe85777bf3b38293b1a18d277507d4a17805fbde62837df592b2">+41/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>servciemonitor.yaml</strong><dd><code>Add ServiceMonitor for Prometheus Exporter</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

charts/kubescape-operator/templates/node-agent/servciemonitor.yaml
- Added a ServiceMonitor for Prometheus if enabled.

</details>
    

  </td>
  <td><a href="https://github.com/kubescape/helm-charts/pull/416/files#diff-99f1149fe7f2152d689ac95d5d06c066e39c6bd7209115f11eae8419f64e0e44">+18/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>service.yaml</strong><dd><code>Introduce Service for Node-Agent</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

charts/kubescape-operator/templates/node-agent/service.yaml
- Introduced a new service for node-agent.

</details>
    

  </td>
  <td><a href="https://github.com/kubescape/helm-charts/pull/416/files#diff-2e8fb76356d730f28a88941cfb4a0d353a3a5855c119a9eebf7e9e22878ec14c">+18/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>configmap.yaml</strong><dd><code>Update Synchronizer Configuration for Runtime Alerts</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

charts/kubescape-operator/templates/synchronizer/configmap.yaml
- Updated synchronizer configuration for runtime alerts.

</details>
    

  </td>
  <td><a href="https://github.com/kubescape/helm-charts/pull/416/files#diff-dbee05c4ed5af6518fa52ee9d2ded91a6c73640791e1a098f2be22a1f9707a27">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr><tr><td><strong>Bug_fix</strong></td><td><details><summary>3 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>clusterrole.yaml</strong><dd><code>Update ClusterRole for Kubescape</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

charts/kubescape-operator/templates/kubescape/clusterrole.yaml
- Updated resources to reflect removal of summaries.

</details>
    

  </td>
  <td><a href="https://github.com/kubescape/helm-charts/pull/416/files#diff-9a6940214e2738d47e402993ba297e091d850319390b5f424f1e1ef8c0ef4d27">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>clusterrole.yaml</strong><dd><code>Update ClusterRole for KubeVuln</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

charts/kubescape-operator/templates/kubevuln/clusterrole.yaml
- Updated resources to reflect removal of summaries.

</details>
    

  </td>
  <td><a href="https://github.com/kubescape/helm-charts/pull/416/files#diff-7ae6fcf55f65837aa45ec5a20d9dcccdc13a70a54e85baf416e1e81f9fcb4a30">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>clusterrole.yaml</strong><dd><code>Update Operator ClusterRole</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

charts/kubescape-operator/templates/operator/clusterrole.yaml
- Updated resources to reflect removal of summaries.

</details>
    

  </td>
  <td><a href="https://github.com/kubescape/helm-charts/pull/416/files#diff-319db9f5c11e560155712379c80232e880c8bafae3f91b6a7ac8e1d2336eb42b">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: charts/kubescape-operator/values.yaml [hunk 1]
```diff
@@ -450,15 +452,19 @@ nodeAgent:
   name: node-agent
   image:
     # -- source code: https://github.com/kubescape/node-agent
-    repository: quay.io/kubescape/node-agent
-    tag: v0.2.15
+    repository: quay.io/amirm_armo/node-agent
+    tag: kdr-alpha
```
Comment (amirmalka) [inline charts/kubescape-operator/values.yaml:456]: TODO: update tag with release @dwertent

File: charts/kubescape-operator/templates/node-agent/daemonset.yaml [hunk 2]
```diff
@@ -77,6 +82,30 @@
...
+              - "echo 'PING' | nc -w 1 localhost 3310 | grep PONG || exit 1"
+            initialDelaySeconds: 50
+            periodSeconds: 10
+          resources:
+{{ toYaml .Values.clamav.resources | indent 12 }}
+          {{- if .Values.clamav.volumeMounts }}
+          volumeMounts:
+            {{- toYaml .Values.clamav.volumeMounts | nindent 10 }}
+          {{- end }}
+        {{- end }}
```
[the remark this answers is not in the document]
Comment (dwertent) [reply charts/kubescape-operator/templates/node-agent/daemonset.yaml:108]: @amitschendel Can you please check for readiness and liveness probs?

File: charts/kubescape-operator/README.md [hunk 3]
```diff
@@ -28,22 +28,12 @@ kubescape     storage-59567854fd-hg8n8                            1/1     Runnin
 
 The scanning results will be available gradually as the scans are completed.
 
-View your configuration scan summaries:
```
Comment (dwertent) [inline charts/kubescape-operator/README.md:31]: @matthyx we should revert both changes

Comment (dwertent) [reply charts/kubescape-operator/README.md:31]: Was this fully removed from the storage?

File: charts/kubescape-operator/templates/node-agent/daemonset.yaml [hunk 4]
```diff
@@ -97,6 +124,10 @@ spec:
             - name: OTEL_COLLECTOR_SVC
               value: "otel-collector:4317"
             {{- end }}
+            {{- if $components.clamAV.enabled }}
+            - name: CLAMAV_ADDRESS
+              value: "127.0.0.1:3310"
```
Comment (dwertent) [inline charts/kubescape-operator/templates/node-agent/daemonset.yaml:129]: TODO: The port should be configured

File: charts/kubescape-operator/templates/node-agent/default-rule-binding.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  # exclude K8s system namespaces
+    matchExpressions:
+      - key: "kubernetes.io/metadata.name"
+        operator: "NotIn"
+        values:
+        - "kube-system"
+        - "kube-public"
+        - "kube-node-lease"
+        - "kubeconfig"
+  # podSelector:
```
Comment (dwertent) [inline charts/kubescape-operator/templates/node-agent/default-rule-binding.yaml:17]: TODO: Remove

File: charts/kubescape-operator/templates/node-agent/servciemonitor.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+# If prometheus enabled, add prometheus exporter
+{{- if eq .Values.nodeAgent.config.prometheusExporter "enable" }}
```
Comment (dwertent) [inline charts/kubescape-operator/templates/node-agent/servciemonitor.yaml:2]: TODO: Verify exporter works

File: charts/kubescape-operator/README.md (hunk 3 above)
Comment (matthyx) [reply charts/kubescape-operator/README.md:31]: configurationscansummaries and vulnerabilitysummaries are still here - but they are virtual objects

File: charts/kubescape-operator/templates/node-agent/daemonset.yaml (hunk 4 above)
Comment (matthyx) [reply charts/kubescape-operator/templates/node-agent/daemonset.yaml:129]: we keep the sidecar?

**Review by yuleib — APPROVED** [review]

File: charts/kubescape-operator/values.yaml [hunk 7]
```diff
@@ -451,14 +453,18 @@ nodeAgent:
...
     pullPolicy: IfNotPresent
 
   config:
-    maxLearningPeriod: 3h # duration string
+    maxLearningPeriod: 12h # duration string
     learningPeriod: 2m # duration string
     updatePeriod: 10m # duration string
+    prometheusExporter: disable
+    httpExporterConfig: 
+      url: http://synchronizer.kubescape.svc.cluster.local:8089/apis/v1/kubescape.io/v1/runtimealerts
```
Comment (amirmalka) [inline charts/kubescape-operator/values.yaml:465]: what if the namespace is not `kubescape`?

File: charts/kubescape-operator/templates/node-agent/daemonset.yaml (hunk 4 above)
Comment (dwertent) [reply charts/kubescape-operator/templates/node-agent/daemonset.yaml:129]: For now, yes.

File: charts/kubescape-operator/README.md (hunk 3 above)
Comment (dwertent) [reply charts/kubescape-operator/README.md:31]: It looks like the storage is not creating them...
```
% kubectl get vulnerabilitymanifestsummaries -A
No resources found
```

Comment (dwertent) [reply charts/kubescape-operator/README.md:31]: And I'm running the latest tag.

File: charts/kubescape-operator/templates/node-agent/configmap.yaml [hunk 8]
```diff
@@ -14,9 +14,27 @@ data:
     {
         "applicationProfileServiceEnabled": {{ $configurations.runtimeObservability }},
         "relevantCVEServiceEnabled": true,
+        "prometheusExporterEnabled": "{{ eq .Values.nodeAgent.config.prometheusExporter "enable" }}",
+        "runtimeDetectionEnabled": "{{ eq .Values.capabilities.runtimeDetection "enable" }}",
```
Comment (amirmalka) [inline charts/kubescape-operator/templates/node-agent/configmap.yaml:18]: We should fix `runtimeDetectionEnabled` and `networkServiceEnabled` to boolean like other existing boolean configs.


```
	EnableFullPathTracing    bool                      `mapstructure:"fullPathTracingEnabled"`
	EnableApplicationProfile bool                      `mapstructure:"applicationProfileServiceEnabled"`
	EnablePrometheusExporter bool                      `mapstructure:"prometheusExporterEnabled"`
	EnableRuntimeDetection   bool                      `mapstructure:"runtimeDetectionEnabled"`
	EnableNetworkTracing     bool                      `mapstructure:"networkServiceEnabled"`
	EnableRelevancy          bool                      `mapstructure:"relevantCVEServiceEnabled"`
```

File: charts/kubescape-operator/templates/node-agent/configmap.yaml [hunk 9]
```diff
@@ -14,9 +14,28 @@ data:
     {
         "applicationProfileServiceEnabled": {{ $configurations.runtimeObservability }},
         "relevantCVEServiceEnabled": true,
+        "prometheusExporterEnabled": {{ .Values.nodeAgent.serviceMonitor.enabled }},
+        "runtimeDetectionEnabled": {{ eq .Values.capabilities.runtimeDetection "enable" }},
+        "networkServiceEnabled": {{ eq .Values.capabilities.networkPolicyService "enable" }},
+        "malwareDetectionEnabled": {{ eq .Values.capabilities.malwareDetection "enable" }},
```
Comment (amirmalka) [inline charts/kubescape-operator/templates/node-agent/configmap.yaml:20]: @dwertent TODO: update node agent tag which supports this new config

Comment (dwertent) [reply charts/kubescape-operator/templates/node-agent/configmap.yaml:20]: Done

File: charts/kubescape-operator/values.yaml (hunk 7 above)
Comment (dwertent) [reply charts/kubescape-operator/values.yaml:464]: It's a good point.
We will change this in the next release

**Review by yuleib — APPROVED** [review]

**Pull request merged**